### DISCUSSION TITLE
What is the period of the perlin noise?

### DISCUSSION THREAD

**Discussion by emilk** [Razaekel/noise-rs#264] [Q&A]
Does the Perlin noise have a finite period length, i.e. does it repeat at a certain distance?

Put another way, is there a `P` such that `perlin.get([x, y]) = perlin.get([x + P, y])` for all `x` and `y`?

**Comment by mitchmindtree** [comment]
I don't believe perlin noise is cycling by default, though if you have a search there are ways you can make it cyclic while remaining smooth. This normally involves sampling the noise function twice at the beginning and end of some period `t` and interpolating between the two.
